### PR TITLE
Updates dashboard organization dropdown options 

### PR DESCRIPTION
**Pull request by jayjaybunce**
#  #

## 🗣 Description ##

Updates which dropdown options are displayed based on userType. 

For global admin, all organizations and tags are displayed in the dropdown along with an "All Organizations" option

For regional admin, only organizations within their region are displayed.

## 💭 Motivation and context ##

Resolves [CRASM-321](https://maestro.dhs.gov/jira/browse/CRASM-321)

## 🧪 Testing ##

Test Case 1 => globalAdmin is able to see all organizations/tags and "All Orgs" options in the dropdown
Test Case 2 => regionalAdmin is able to see only orgs in the users region
Test Case 3 => standard user is not able to see the dropdown

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: frontend/src/components/Header.tsx [hunk 1]
```diff
@@ -199,12 +198,14 @@ const HeaderNoCtx: React.FC<ContextType> = (props) => {
...
-      let tags: (OrganizationTag | Organization)[] = [];
+      let tags: OrganizationTag[] = [];
       if (userLevel === GLOBAL_ADMIN) {
         tags = await apiGet<OrganizationTag[]>('/organizations/tags');
         await setTags(tags as OrganizationTag[]);
       }
-      await setOrganizations(tags.concat(rows));
+      // Not sure why we need to add tags to the orgs array, will look for the purpose. Commenting out for now to fix type union issue
+      await setOrganizations(rows);
+      // await setOrganizations(rows)
```
Comment (cduhn17) [inline frontend/src/components/Header.tsx]: Is this an intended duplicate?

**Review by cduhn17 — COMMENTED** [review]
See comment

File: frontend/src/components/Header.tsx (hunk 1 above)
Comment (jayjaybunce) [reply frontend/src/components/Header.tsx]: Unintended - removed

**Review by schmelz21 — APPROVED** [review]

**Review by cduhn17 — APPROVED** [review]
LGTM

**Pull request merged**